packages/core/src/browser/messaging/ws-connection-provider.ts

```diff
@@ -61,28 +91,93 @@ export class WebSocketConnectionProvider extends AbstractConnectionProvider<WebS
             this.fireSocketDidClose();
         };
         socket.onmessage = ({ data }) => {
+            this.httpFallbackCounter = 0;
             this.handleIncomingRawMessage(data);
         };
         this.socket = socket;
         window.addEventListener('offline', () => this.tryReconnect());
         window.addEventListener('online', () => this.tryReconnect());
     }
 
+    handleSocketError(event: unknown): void {
+        this.httpFallbackCounter += 1;
+        if (this.httpFallbackOptions.allowed && this.httpFallbackCounter >= this.httpFallbackOptions.trigger) {
+            this.useHttpFallback = true;
+            this.socket.close();
+            const httpUrl = this.createHttpWebSocketUrl(WebSocketChannel.wsPath);
+            this.onHttpFallbackDidActivateEmitter.fire(undefined);
+            this.longPolling(httpUrl);
+            this.messageService().warn(
+                'Could not establish a websocket connection. The application will be using the HTTP fallback mode. This may affect performance and the behavior of some features.'
+            );
+        }
+        console.error(event);
+    }
+
+    async longPolling(url: string): Promise<void> {
```

Comment:
```suggestion
    async initiatePolling(url: string): Promise<void> {
```
Or some other verb?

Reply:
I used `doLongPolling` now, since every call to this method only sets up a single long polling call which repeats itself using a timeout.